master/internal/api_tasks.go

```diff
@@ -429,10 +430,24 @@ func constructTaskLogsFilters(req *apiv1.TaskLogsRequest) ([]api.Filter, error)
 		}
 	}
 
+	// Allow a value in a list of numbers, or a NULL represented as -1.
+	addNullInclusiveFilter := func(field string, values []int32) {
+		if values == nil || !slices.Contains(values, -1) {
+			addInFilter(field, values, len(values))
+			return
+		}
+		filters = append(filters, api.Filter{
+			Field:     field,
+			Operation: api.FilterOperationInOrNull,
```

Comment:
Remove the -1 from values?

Reply:
✅

Reply:
ok, actually undid this as there were some issues with the implementation.  Since rank=-1 will not match items, it's not harmful. To remove the -1, we can't have one line to remove a value from a slice, instead we would need to build a new slice and use a for loop to replace any -1 with the last item, it just was messier than expected.